src/__tests__/searchAlgorithm.test.js

```diff
@@ -0,0 +1,19 @@
+import searchAlgorithm from '../utils/searchAlgorithm'
+describe('searchAlgorithm', () => {
+  it('returns filtered results when input and data are provided', () => {
+    const input = 'mass'
+    const colleges = [{ name: 'Massachusetts Institute of Technology (MIT)', country: 'United States', city: 'Cambridge' }]
+    const result = searchAlgorithm(input, colleges)
+    expect(result).toEqual([{ name: 'Massachusetts Institute of Technology (MIT)', country: 'United States', city: 'Cambridge' }])
+  })
+  it('returns an empty array when either input is not provided', () => {
+    const input = ''
+    const resultWithoutData = searchAlgorithm(input, null)
```

Comment:
Let's separate the setup lines, act and assertions by adding blank lines below and above the act.

Reply:
You might want to rename the variable `resultWithoutData` to `result`, once you decide to use the `colleges` variable as data.